### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/builtins/BuiltinArray.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ Value builtinArrayConstructor(ExecutionState& state, Value thisValue, size_t arg
 
     if (interpretArgumentsAsElements) {
         Value val = argv[0];
-        if (argc > 1 || !val.isInt32()) {
+        if (argc > 1 || !val.isNumber()) {
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/builtins/BuiltinArray.cpp:68]: That's a good idea!

File: src/runtime/EncodedValue.h [hunk 2]
```diff
@@ -326,53 +326,59 @@ class EncodedValue {
...
         if (from.isInt32() && EncodedValueImpl::PlatformSmiTagging::IsValidSmi(i32 = from.asInt32())) {
             m_data.payload = EncodedValueImpl::PlatformSmiTagging::IntToSmi(i32);
-            return;
+            return *this;
         }
 
         if (from.isNumber()) {
+            Value mutableFrom(from);
+            if (UNLIKELY(Value::isInt32ConvertibleDouble(mutableFrom.asNumber(), i32))) {
+                mutableFrom = Value(i32);
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/EncodedValue.h:360]: I had tested two way
a. as is
b. you mentioned
but 'a' method is faster. especially crypto test

IMHO I think it is possible.
once double type value saved,
double type value is more likely to be stored in the place

File: src/runtime/EncodedValue.h [hunk 3]
```diff
@@ -326,53 +326,59 @@ class EncodedValue {
         return operator Escargot::Value().toUint32(state);
     }
 
-    void operator=(PointerValue* from)
+    const EncodedValue& operator=(PointerValue* from)
     {
         ASSERT(from);
         m_data.payload = (intptr_t)from;
+        return *this;
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/EncodedValue.h:333]: There is no specific reason for this
but many operator= function returns const reference of this

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**